templates/ra_guac_private_autoscale_elb.element

```diff
@@ -142,13 +148,22 @@
         "UseURLText2" :
         {
             "Fn::Not" : [ { "Fn::Equals" : [ { "Ref" : "URLText2" }, "" ] } ]
-        }
+        },
+        "UseAmiId" :
+        {
+            "Fn::Not" : [ { "Fn::Equals" : [ { "Ref" : "AmiId" }, "" ] } ]
+        },
+        "NotToUseAmiId" :
+        {
+              "Fn::Equals" : [ { "Ref" : "AmiId" }, "" ] 
+        }		
```

Comment:
Ahhh, ok, this _can_ be collapsed to a single condition, just need to reverse the `If` logic. But first, rename `NotToUseAmiID` to `UseAmiLookup`. In the custom `AmiIdLookup` resource, nothing changes (except the renamed parameter). Then in the `Fn::If` statement, reverse the logic. E.g.

```
               "ImageId" :
               {
                   "Fn::If" :
                   [
                           "UseAmiLookup",
                           { "Fn::GetAtt" : [ "AmiIdLookup", "Id" ] },
                           { "Ref" : "AmiId" }
                   ]
               },
```

Take note of the whitespace above. Trying to standardize a bit. Braces and brackets should be on new lines, except where the entire object fits within a 79 char line, including the closing brace/bracket. Four space indents. No trailing whitespace at the end of a line. Single space before and after a colon, single space after opening brace/bracket, single space before closing brace/bracket (except when such characters are at the end of the line, no trailing whitespace takes precedent).